torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -354,8 +354,9 @@ def load(self, example_inputs, fx_config: _CompileFxKwargs) -> CompiledFxGraph:
             payload_fn=lambda: json.dumps(cache_info),
         )
 
-        FxGraphCache.post_compile(result, example_inputs, fx_config["cudagraphs"])  # type: ignore[arg-type]
-        result._boxed_call = True
+        # TODO: How come cudagraphs could be None here?
+        # TODO: How come gm is None here?
+        result.post_compile(example_inputs, fx_config["cudagraphs"], None)  # type: ignore[arg-type]
```

Comment:
gm is only used for @masnesral's specific freezing use case. In that case, we need the post compile gm to do things. But we don't have that for AOTAutogradCache, so AOTAutogradCache doesn't support the freezing workflow. It always passes in None here since the gm should never be needed for post compile.

(The only "gm" it has access to at this step is the post dynamo gm. So anything needed to load from FXGraphCache needs to be obtainable from that)

Reply:
Hmm, are we talking about the same code? I see here in `post_compile`:

```
        if cudagraphs:
            # It's possible that cudagraphs is enabled, but was disabled
            # during a previous compilation we're loading from the cache.
            # If so, we need to disable it on this new process too.
            if self.disabled_cudagraphs_reason:
                if "cuda" in self.device_types:
                    log_cudagraph_skip_and_bump_counter(
                        f"skipping cudagraphs due to {self.disabled_cudagraphs_reason}"
                    )
                else:
                    counters["inductor"]["cudagraph_skips"] += 1
                BoxedBool.disable(cudagraphs)
            else:
                cudagraph_post_compile(
                    example_inputs,
                    self,
                    cudagraphs,
                    gm,
                )
```

So gm is needed to do some CUDA graphs stuff. I don't see how AOTAutograd cache proves that cudagraphs could never be enabled here. I can try to write a test case that tickles it... later, maybe.

Reply:
Ah yeah, it's confusing because cudagraphs_post_compile only uses gm if freezing was enabled on the cache write, and CompiledFxGraph doesn't have the constants on hand already (which it always will in AOTAutogradCache scenarios) 

IMO we should change this so the gm is not passed to cudagraphs_post_compile: only the constants themselves, and instead have whatever caller to cudagraphs_post_compile handle obtaining the constants from the gm. 

The gm field is really confusing to work with because it is essentially untyped: it can represent both a dynamo gm and a post aotdispatch gm.

Reply:
@masnesral It's really confusing what's happening here so I might refactor it some more: we shouldn't call CompoledFxGraph.get_constants() with an optional gm and change its behavior depending on some unspoken invariants. We should only call this helper function *if* we know freezing is on in the cache entry and we're not coming from AOTautogradcache. 

The way it's written now there's no way to tell that this gm is always None from the exact code path where it's precisely correct, and a correct value otherwise